pg_search/src/postgres/customscan/pdbscan/exec_methods/top_n.rs

```diff
@@ -281,22 +282,22 @@ impl ExecMethod for TopNScanExecState {
 
         // Run the TopN (and optional aggregate) query.
         self.search_results = if let Some(orderby_info) = self.orderby_info.as_ref() {
-            let maybe_aggregation_collector = aggregations.as_ref().map(|aggregations| {
+            let maybe_aux_collector = aggregations.as_ref().map(|aggregations| {
                 // Wrap the aggregation collector with MVCC filtering to respect transaction visibility.
                 // This ensures that aggregations only include documents visible to the current transaction,
                 // matching the behavior of the TopN results.
-                let base_collector = DistributedAggregationCollector::from_aggs(
-                    aggregations.aggregations.clone(),
-                    agg_limits.clone(),
-                );
-
                 let heaprel = state.heaprel();
-                MVCCFilterCollector::new(
-                    base_collector,
-                    TSVisibilityChecker::with_rel_and_snap(heaprel.as_ptr(), unsafe {
-                        pg_sys::GetActiveSnapshot()
-                    }),
-                )
+                TopNAuxiliaryCollector {
+                    aggregation_collector: DistributedAggregationCollector::from_aggs(
+                        aggregations.aggregations.clone(),
+                        agg_limits.clone(),
+                    ),
+                    // TODO: Expose an option to disable this.
```

Comment:
This TODO will be a followup by @mdashti or myself.

Reply:
```suggestion
                    // TODO: Expose a query-time GUC or function parameter to disable MVCC filtering
                    // for performance in cases where the visibility map is known to be up-to-date.
```